primitives/transaction/src/historic_transaction.rs

```diff
@@ -159,15 +159,33 @@ impl HistoricTransaction {
         }
     }
 
-    /// Returns the hash of the underlying transaction/inherent. For reward inherents we return the
-    /// hash of the corresponding reward transaction. This results into an unique hash for the
-    /// reward inherents (which wouldn't happen otherwise) and allows front-end to fetch rewards by
-    /// their transaction hash.
+    /// Returns the hash based on the underlying transaction/event.
     pub fn tx_hash(&self) -> Blake2bHash {
         match &self.data {
             HistoricTransactionData::Basic(tx) => tx.hash(),
-            // TODO: check for freedom of hash prefixes
-            _ => self.data.hash(),
+            _ => {
+                let block_height = if !matches!(self.data, HistoricTransactionData::Equivocation(_))
+                {
+                    self.block_number
+                } else {
+                    // Equivocations are not distinguished by block height.
+                    0
+                };
```

Comment:
I'd turn this condition around. It makes more sense if the if block handles the match with the specifically mentioned data type, and the else block handles all other cases.

```suggestion
                let block_height = if matches!(self.data, HistoricTransactionData::Equivocation(_))
                {
                    // Equivocations are not distinguished by block height.
                    0
                } else {
                    self.block_number
                };
```

Reply:
Done.